tools/autograd/derivatives.yaml

```diff
@@ -1532,11 +1532,11 @@
   result: auto_linear
 
 - name: sum.SymInt(Tensor self, SymInt[1] dim, bool keepdim=False, *, ScalarType? dtype=None) -> Tensor
-  self: sum_backward(grad, self.sym_sizes(), dim, keepdim)
+  self: sum_backward_symint(grad, self.sym_sizes(), dim, keepdim)
   result: auto_linear
 
 - name: sum.dim_IntList(Tensor self, int[1]? dim, bool keepdim=False, *, ScalarType? dtype=None) -> Tensor
```

Comment:
@albanD  do you have any thoughts on what is the best way to register a NestedTensor kernel for a backward formula that accepts `self.sizes()`? It seems that this is the case for most of the reductions :( 

Right now I'm doing something super hacky where I add a kwarg self to sum_backward and pass a dummy in place of sizes which is probably really bad. I recall Driss mentioned something about editing VariableTypeManual.cpp but you mentioned yesterday that we probably shouldn't do this, is there any other way around this?